### ISSUE TITLE
Add support for Mike hidden version flag

### ISSUE THREAD

**Issue by lcollins** [squidfunk/mkdocs-material#6904]
### Description

We have MkDocs using Material within our organisation. We have been using the Mike plugin to publish the versions using SemVer 0.20.0 etc but equally we also publish the HEAD of the release branch as dev-0.20.x. Currently both versions are visible in the version dropdown. 

We want to hide the dev-0.20.x versions and hoping we can do so using a feature the Mike maintainer mentioned on a similar issue https://github.com/jimporter/mike/issues/151#issuecomment-1806507108

### Related links

- [Hidden version feature](https://github.com/jimporter/mike/issues/151#issuecomment-1806507108)

### Use Cases

We would like versions with the hidden property be excluded from the version selector

Here is an example of the version.json file 

``` json
[
  {
    "version": "dev-0.20.x",
    "title": "dev-0.20.x",
    "aliases": [
      "0.20.x"
    ],
    "properties": {
      "hidden": true
    }
  },
  {
    "version": "dev-main",
    "title": "dev-main",
    "aliases": [
      "development-latest",
      "main"
    ],
    "properties": {
      "hidden": true
    }
  },
  {
    "version": "1.0.0",
    "title": "1.0.0",
    "aliases": [
      "latest"
    ]
  },
  {
    "version": "0.20.1",
    "title": "0.20.1",
    "aliases": []
  },
  {
    "version": "0.20.0",
    "title": "0.20.0",
    "aliases": []
  }
]
```

### Before submitting

- [X] I have read and followed the [change request guidelines](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/).
- [X] I have verified that [my idea is a change request and not a bug report](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/#its-not-a-bug-its-a-feature).
- [X] I have ensured that, to the best of my knowledge, [my idea will benefit the entire community](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/#benefit-for-the-community).
- [X] I have included relevant links to [the documentation](https://squidfunk.github.io/mkdocs-material/), related [issues](https://github.com/squidfunk/mkdocs-material/issues), and [discussions](https://github.com/squidfunk/mkdocs-material/discussions) to underline the need for my idea.

**Comment by squidfunk** [comment]
Thanks for suggesting. Without diving into the details, I think it is reasonable and simple to support ☺️

**Comment by squidfunk** [comment]
Added in 2f1b2e950. If a version has the `hidden` property, it will not be rendered in the selector. Everything else should keep working as-is.

**Comment by lcollins** [comment]
Wow that was fast, thank you for your support @squidfunk

**Comment by squidfunk** [comment]
Released as part of [9.5.14](https://squidfunk.github.io/mkdocs-material/changelog/#9.5.14).
